### ISSUE TITLE
No routing along multipolygons

### ISSUE THREAD

**Issue by scaidermern** [osmandapp/Osmand#8438]
Foot routing along the edge of a `highway=pedestrian` multipolygon is not possible:

![photo_2020-02-11_11-58-39](https://user-images.githubusercontent.com/1040784/74231228-68943380-4cc6-11ea-8f96-85dc7804cbc3.jpg)

Start: 51.0725174,13.7567825
End: 51.0725829,13.7571788
Route: [route.gpx.txt](https://github.com/osmandapp/Osmand/files/4185258/route.gpx.txt)
Relation: https://www.openstreetmap.org/relation/3937735

**Comment by scaidermern** [comment]
> Might even be a manifestation of [#3947](https://github.com/osmandapp/OsmAnd/issues/3947).

No. This multipolygon is not an old-style multipolygon. The tags are correctly on the relation, not on the outer way.

**Comment by dreirund** [comment]
> > Might even be a duplicate of [#3947](https://github.com/osmandapp/OsmAnd/issues/3947).
> 
> *No. This multipolygon is not an old-style multipolygon. The tags are correctly on the relation, not on the outer way.*

Hm.

In my understanding, #3947 is about that old-style multipolygons are deprecated, and routing engine cannot route on (the edges of) "new-style" multipolygons.  
As this example here correctly is one "new-style" multipolygon which is not considered for routing.

**Comment by scaidermern** [comment]
Maybe old-style multipolygons are better suited for routing but they had a bunch of other problems. This here is really something that needs to be fixed in routing engines, not in the map data. Maybe some preprocessing during map generation is enough to make them routeable.
